packages/react-components/src/components/RemoteVideoTile.tsx

```diff
@@ -39,27 +39,12 @@ export const RemoteVideoTile = React.memo(
       showMuteIndicator
     } = props;
 
-    useEffect(() => {
-      if (isAvailable && !renderElement) {
-        onCreateRemoteStreamView && onCreateRemoteStreamView(userId, remoteVideoViewOption);
-      }
-      if (!isAvailable) {
```

Comment:
I think I'm still confused on the solution here, from the PR desc, you mention that onDisposeRemoteStreamView being called is the reason for the recreation of the tile - but why does this get called?
Is it being called here when `!isAvailable` or is it being called in the `return () => { onDisposeRemoteStreamView && onDisposeRemoteStreamView(userId);`?

Reply:
Sorry for the confusion. I updated the description. 
To answer the quesiton, it is being called in the second useEffect with `return () => { onDisposeRemoteStreamView && onDisposeRemoteStreamView(userId);` after the `RemoteVideoTile` at the end of its lifetime when the HorizontalGallery page changes. So when the screensharer's `RemoteVideoTile` ends its lifetime this causes a change in the VideoGalleryRemoteParticipant object of the screensharer. Since the VideoGalleryRemoteParticipant object of the screensharer is a prop of `RemoteScreenShare`, it will cause a rerender because of the change.

Reply:
My concern here is now, at the end of this tile's lifetime, onDisposeRemoteStreamView is not being called, so ultimately `callClient.disposeView` is never being called. Not sure what any implications that actually has, but its probably a resource we should be cleaning up.

`onDisposeRemoteStreamView ` is super unopinionated and cleans up both the remote camera stream and the screenshare stream:
![image](https://user-images.githubusercontent.com/2684369/143133487-a404b76c-2d03-4c98-89cf-c4d873e3458e.png)

Given we might want to create just a screenshare stream or a video feed independently (and in this case - just dispose a video feed and not a screenshare stream) - I think we need to distinguish in onCreateRemoteStreamView and onDisposeRemoteStreamView if we actually want both or just one or the other.

+ @PorterNan for opinion here

This'll likely be a change to the API, so if this change doesn't cause any performance issues (e.g. imagine a video of 40+ people, if they've each had a video feed created at some point but never disposed - will we hit problems?) I'm happy to keep this change and file a work item for after GA to handle videoFeed and screenshareFeed seperately

Reply:
Thanks for pointing out that `onDisposeRemoteStreamView ` disposes both video stream and screenshare stream. I was not aware of that. That is the main reason for the re-render.